### DISCUSSION TITLE
Adding Cascade referential action to onDelete on existing schema relationships

### DISCUSSION THREAD

**Discussion by cordial** [prisma/prisma#18576] [Q&A] [answered]
If you add cascading deletes for a @relation to an existing schema (with data), and then run / create a migration - do you manually need to edit the migration file or is is sufficient to create the migration and then any old data will use the new referential action without any foreign constraint violation issues?

**Comment by ludralph** [comment, marked as answer]
Hi @cordial 👋 

When you add cascading deletes to a `@relation` in an existing schema, the next step is to create a migration to apply these changes to the database. If you have existing data in the database, you will need to ensure that the migration does not violate any foreign key constraints. If the existing data violates the foreign key constraints that will be affected by the cascading delete, the migration will fail with an error indicating that the data needs to be fixed before the migration can be applied. In that case, you will need to manually fix the data to ensure that it complies with the new constraints before running the migration.

**Reply by ludralph** [reply]
Hi @cordial 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Comment by joshflint** [comment]
It would be great if Prisma could automatically generate a line for updating the referential action in SQL. We've been tripped up multiple times by assuming a "onDelete" in the schema was applied to our DB, when really no one ever created the mirgation.

Prisma can do the SQL migration for new relations - it would be great if Prisma could also update it for existing relations when there's a change.

**Reply by janpio** [reply]
Doesn't it do that? I was 100% under the impression that _any_ relevant change in Prisma schema file would lead to a migration. If not, that is a bug.
